src/components/InvertedFlatList/BaseInvertedFlatList.js

```diff
@@ -52,60 +46,66 @@ class BaseInvertedFlatList extends Component {
+                // initialRowHeight since we can only assume that all previous items
+                // have not yet been measured
+                offset: _.isUndefined(lastMeasuredItem) ? initialRowHeight * index : lastMeasuredItem.offset + initialRowHeight,
                 index,
             };
-        }
-
-        // If we don't have a size yet means we haven't measured this
-        // item yet. However, we can still calculate the offset by looking
-        // at the last size we have recorded (if any)
-        const lastMeasuredItem = CollectionUtils.lastItem(this.sizeMap);
-
-        return {
-            // We haven't measured this so we must return the minimum row height
-            length: this.props.initialRowHeight,
-
-            // Offset will either be based on the lastMeasuredItem or the index +
-            // initialRowHeight since we can only assume that all previous items
-            // have not yet been measured
-            offset: _.isUndefined(lastMeasuredItem) ? this.props.initialRowHeight * index : lastMeasuredItem.offset + this.props.initialRowHeight,
-            index,
-        };
-    }
+        },
+        [initialRowHeight, sizeMap],
+    );
 
     /**
      * Measure item and cache the returned length (a.k.a. height)
      *
      * @param {React.NativeSyntheticEvent} nativeEvent
      * @param {Number} index
      */
-    measureItemLayout(nativeEvent, index) {
-        const computedHeight = nativeEvent.layout.height;
-
-        // We've already measured this item so we don't need to
-        // measure it again.
-        if (this.sizeMap[index]) {
-            return;
-        }
-
-        const previousItem = this.sizeMap[index - 1] || {};
-
-        // If there is no previousItem this can mean we haven't yet measured
-        // the previous item or that we are at index 0 and there is no previousItem
-        const previousLength = previousItem.length || 0;
-        const previousOffset = previousItem.offset || 0;
-        this.sizeMap[index] = {
-            length: computedHeight,
-            offset: previousLength + previousOffset,
-        };
-    }
+    const measureItemLayout = useCallback(
+        (nativeEvent, index) => {
+            const computedHeight = nativeEvent.layout.height;
+
+            // We've already measured this item so we don't need to
+            // measure it again.
+            if (sizeMap[index]) {
+                return;
+            }
+
+            const previousItem = sizeMap[index - 1] || {};
+
+            // If there is no previousItem this can mean we haven't yet measured
+            // the previous item or that we are at index 0 and there is no previousItem
+            const previousLength = previousItem.length || 0;
+            const previousOffset = previousItem.offset || 0;
+            sizeMap[index] = {
+                length: computedHeight,
+                offset: previousLength + previousOffset,
+            };
+        },
+        [sizeMap],
```

Comment:
`sizeMap` because it is a ref it won't produce a rerender when it changes, so you can remove it from dependencies

Reply:
@gedu If i remove sizeMap here, lint will definitely yell at it because ```measureItemLayout``` is a dependency in the ```renderItemFromProp```. Two things i could do here, i could disable lint warning - or entirely remove all memo. what do you suggest?